crates/re_data_store/src/store_db.rs

```diff
@@ -318,6 +329,8 @@ impl StoreDb {
         // cascades, thus this whole process must stabilize after one iteration.
         debug_assert!(clear_cascade.is_empty());
 
+        self.stats.on_new_row_id(row_id);
```

Comment:
the has the look and feel of a store subscriber. Can we make it a store subscriber? That way you also can put it somewhere als and don't need to drag new dependencies into re_data_store!